jetty-client/src/main/java/org/eclipse/jetty/client/HttpChannel.java

```diff
@@ -103,10 +105,7 @@ public boolean disassociate(HttpExchange exchange)
 
     public HttpExchange getHttpExchange()
     {
-        synchronized (this)
-        {
-            return _exchange;
-        }
+        return _lock.runLocked(() -> _exchange);
```

Comment:
Can we verify if this style does more allocations or not?
```suggestion
        try (AutoLock ignored = _lock.lock())
        {
            return _exchange;
        }		￼   
```
is not that bad.

Reply:
AFAIK, `_lock.runLocked(() -> _exchange)` must capture `this` and therefore allocates each time.

It may or may not be optimized by allocating only the first time it's executed, however the `try` version you suggest does not allocate ever, but it's more boilerplate-y.

Reply:
I don't think we should be allocating for syntax sugar.     If we can't demonstrate that these simple getters/setters are not optimised to not allocate, then I think we should use the boilerplate version.... it's only 3 extra lines.